monkey/infection_monkey/network/relay/tcp_pipe_spawner.py

```diff
@@ -0,0 +1,34 @@
+import socket
+from ipaddress import IPv4Address
+from typing import List
+
+from .tcp import SocketsPipe
+
+
+class TCPPipeSpawner:
```

Comment:
`# TODO: This needs to periodically remove dead pipes from its list.`

Perhaps every `x` times `spawn_pipe()` is called, this can clean up its list.